service/history/execution/mutable_state_builder.go

```diff
@@ -1152,6 +1152,36 @@ func (e *mutableStateBuilder) GetStartEvent(
 	return startEvent, nil
 }
 
+// GetCloseEvent returns the last event in history
+func (e *mutableStateBuilder) GetCloseEvent(
+	ctx context.Context,
+) (*types.HistoryEvent, error) {
+
+	if e.GetExecutionInfo().CloseStatus == persistence.WorkflowCloseStatusNone {
+		return nil, nil
```

Comment:
not sure if it's better to return an error here. Usually caller will assume historyEvent is not nil when error is nil?

Reply:
Will return error when close event is nil